handlers/geolocation/geolocation.go

```diff
@@ -24,38 +25,61 @@ type GeolocationHandlersCollection struct {
 
 // this package handles geolocation for playback and origin discovery for node replication
 
-// redirect an incoming user to a node for playback or 404 handling
+// Redirect an incoming user to the CDN, closest node (geolocate)
+// or another service (like mist HLS) on the current host for playback.
+// nodeHost logic is related 404 handling.
 func (c *GeolocationHandlersCollection) RedirectHandler() httprouter.Handle {
-	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
-		nodeHost := c.Config.NodeHost
-		redirectPrefixes := c.Config.RedirectPrefixes
-
-		if nodeHost != "" {
-			host := r.Host
-			if host != nodeHost {
-				newURL, err := url.Parse(r.URL.String())
-				if err != nil {
-					glog.Errorf("failed to parse incoming url for redirect url=%s err=%s", r.URL.String(), err)
-					w.WriteHeader(http.StatusInternalServerError)
+	return func(w http.ResponseWriter, r *http.Request, params httprouter.Params) {
+		host := r.Host
+		pathType, prefix, playbackID, pathTmpl := parsePlaybackID(r.URL.Path)
+
+		if c.Config.CdnRedirectPrefix != nil && (pathType == "hls" || pathType == "webrtc") {
+			doCdnRedirect := contains(playbackID, c.Config.CdnRedirectPlaybackIDs)
+			if doCdnRedirect {
+				if pathType == "webrtc" {
+					w.WriteHeader(http.StatusNotAcceptable) // 406 error
+					metrics.Metrics.CDNRedirectWebRTC406.WithLabelValues(playbackID).Inc()
+					glog.V(6).Infof("%s not supported for CDN-redirected %s", r.URL.Path, playbackID)
```

Comment:
After deploying on staging it will be tested in the player to make sure that 406 falls back into `/hls/` endpoint.